services/cloudwatch/src/it/java/software/amazon/awssdk/services/cloudwatch/CloudWatchIntegrationTest.java

```diff
@@ -374,6 +377,23 @@ public void testExceptionHandling() throws Exception {
         }
     }
 
+    @Test
+    public void testQueryCompatibleExceptionHandling() {
```

Comment:
nit, can we create unit tests with mock server and mock response instead to make the test more reliable?